avalon/maya/pipeline.py

```diff
@@ -568,6 +568,27 @@ def _on_task_changed(*args):
     if os.path.exists(workdir):
         logger.info("Updating Maya workspace for task change to %s", workdir)
         _set_project()
+
+        # Update file dialog starting dir
+        # (NOTE) using `mel.eval` because this `workspace` Python cmd failed
+        #   in Maya 2017 +
+        frule_scene = mel.eval("workspace -q -fileRuleEntry \"scene\"")
```

Comment:
I think in Python it somehow is:

```python
import maya.cmds as cmds
cmds.workspace(fileRuleEntry="scene")
```

Thus without the query. I think it was different in older versions of Maya where it would actually need the query flag, but haven't tested it.

Thanks for the comment here.

Reply:
So this is the command that fit to all version...
I gave it a test run, and [all passed](https://travis-ci.org/davidlatwe/maya-test/builds/429989661)
I will change to this cmd, much simpler.